sqlglot/dialects/clickhouse.py

```diff
@@ -603,6 +603,22 @@ def _parse_join(
             if join:
                 join.set("global", join.args.pop("method", None))
 
+                # tbl ARRAY JOIN arr <-- this should be a `Column` reference, not a `Table`
+                # https://clickhouse.com/docs/en/sql-reference/statements/select/array-join
+                if join.kind == "ARRAY":
+                    for table in join.find_all(exp.Table):
+                        if not table.db:
```

Comment:
What case is this branch protecting against? Afaict such queries seem valid, where `table.db` is the actual table name

```
:) with t as (select [1, 2, 3] as col, 1 as id) select * from t array join t.col;

WITH t AS
    (
        SELECT
            [1, 2, 3] AS col,
            1 AS id
    )
SELECT *
FROM t
ARRAY JOIN t.col

Query id: 4cef17fc-03e8-4644-a87d-c50f22055034

   ┌─col─┬─id─┐
1. │   1 │  1 │
2. │   2 │  1 │
3. │   3 │  1 │
   └─────┴────┘
```